src/software/ai/passing/pass_evaluation_test.cpp

```diff
@@ -0,0 +1,102 @@
+        Timestamp::fromSeconds(10));
+
+    ASSERT_EQ(pass_eval.getBestPassOnField(),
+              passes_with_rating.at(EighteenZoneId::ZONE_8));
+}
+
+
+TEST(PassEvaluation, best_pass_in_zones)
+{
+    auto field          = Field::createSSLDivisionBField();
+    auto pitch_division = std::make_shared<const EighteenZonePitchDivision>(field);
+
+    std::unordered_map<EighteenZoneId, PassWithRating> passes_with_rating;
+
+    for (EighteenZoneId zone_id : allValuesEighteenZoneId())
+    {
+        passes_with_rating.emplace(zone_id,
+                                   PassWithRating{std::move(Pass(Point(0, 0), 0)),
+                                                  0.05 * static_cast<unsigned>(zone_id)});
+    }
+
+    auto pass_eval = PassEvaluation<EighteenZoneId>(
+        pitch_division, passes_with_rating, std::make_shared<const PassingConfig>(),
+        Timestamp::fromSeconds(10));
+
+    // we expect the zone with the higher id to have the higher score
+    EXPECT_EQ(
+        pass_eval.getBestPassInZones(
+            {EighteenZoneId::ZONE_12, EighteenZoneId::ZONE_10, EighteenZoneId::ZONE_3}),
+        passes_with_rating.at(EighteenZoneId::ZONE_12));
+    EXPECT_EQ(
+        pass_eval.getBestPassInZones(
+            {EighteenZoneId::ZONE_15, EighteenZoneId::ZONE_12, EighteenZoneId::ZONE_13}),
+        passes_with_rating.at(EighteenZoneId::ZONE_15));
+
+    // sanity check that best pass in zones and zone return the same thing when querried
+    // w/ 1 zone
+    EXPECT_EQ(pass_eval.getBestPassInZones({EighteenZoneId::ZONE_10}),
+              passes_with_rating.at(EighteenZoneId::ZONE_10));
+    EXPECT_EQ(pass_eval.getBestPassInZones({EighteenZoneId::ZONE_12}),
+              passes_with_rating.at(EighteenZoneId::ZONE_12));
+    EXPECT_EQ(pass_eval.getBestPassInZones({EighteenZoneId::ZONE_15}),
+              passes_with_rating.at(EighteenZoneId::ZONE_15));
+}
+
+TEST(PassEvaluation, get_pitch_division_and_timestamp)
+{
+    auto field          = Field::createSSLDivisionBField();
+    auto pitch_division = std::make_shared<const EighteenZonePitchDivision>(field);
+
+    std::unordered_map<EighteenZoneId, PassWithRating> passes_with_rating;
+
+    for (EighteenZoneId zone_id : allValuesEighteenZoneId())
+    {
+        passes_with_rating.emplace(zone_id,
+                                   PassWithRating{std::move(Pass(Point(0, 0), 0)), 0});
+    }
+
+    auto pass_eval = PassEvaluation<EighteenZoneId>(
+        pitch_division, passes_with_rating, std::make_shared<const PassingConfig>(),
+        Timestamp::fromSeconds(10));
+
+    EXPECT_EQ(pass_eval.getFieldPitchDivsion(), pitch_division);
+    EXPECT_EQ(pass_eval.getEvaluationTime(), Timestamp::fromSeconds(10));
+}
+
+bool operator==(const PassWithRating &lhs, const PassWithRating &rhs)
+{
+    return lhs.rating == rhs.rating &&
+           lhs.pass.receiverPoint() == rhs.pass.receiverPoint() &&
+           lhs.pass.speed() == rhs.pass.speed();
+}
```

Comment:
why is this defined in the test and not in the class?

Reply:
`PassWithRating` doesn't need the == implemented, its only useful for testing

Reply:
I don't like == being defined outside of the struct/class because there could be multiple implementations

Reply:
I don't see anyone ever needing this == outside of the test, but I don't see any harm if someone did want to use it so moved to `pass_with_rating.cpp`